### PR TITLE
x-pack/filebeat/input/awss3 Add AWS Endpoint resolver

### PR DESCRIPTION
**Pull request by bhapas**
## What does this PR do?

This PR adds `endpoint resolver` to the AWS config.

## Why is it important?

By adding custom endpoint resolver config options to AWSConfig , It becomes flexible to use the input against any `AWS like` resources, for example , when testing against Localstack.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.
- [x] I have made my commit title and message explanatory about the purpose and the reason of the change

### CODE REVIEW

File: x-pack/filebeat/input/awss3/config.go [hunk 1]
```diff
@@ -100,9 +100,9 @@ func (c *config) Validate() error {
 	}
 
 	if c.AWSConfig.FIPSEnabled && c.NonAWSBucketName != "" {
-		return errors.New("fips_enabled cannot be used with a non-AWS S3 bucket.")
+		return errors.New("fips_enabled cannot be used with a non-AWS S3 bucket")
 	}
-	if c.PathStyle && c.NonAWSBucketName == "" {
+	if c.PathStyle && c.NonAWSBucketName == "" && c.QueueURL == "" {
 		return errors.New("path_style can only be used when polling non-AWS S3 services")
```
Comment (kaiyan-sheng) [inline x-pack/filebeat/input/awss3/config.go:106]: maybe edit this error message as well to include queue_url?

File: x-pack/filebeat/input/awss3/config_test.go [hunk 2]
```diff
@@ -408,19 +408,6 @@ func TestConfig(t *testing.T) {
 			expectedErr: "path_style can only be used when polling non-AWS S3 services",
 			expectedCfg: nil,
 		},
-		{
-			name:           "error on path_style with AWS SQS Queue",
```
Comment (kaiyan-sheng) [inline x-pack/filebeat/input/awss3/config_test.go:412]: can we put the deleted test back?

Comment (bhapas) [reply x-pack/filebeat/input/awss3/config_test.go:412]: This test criteria has changed now as we added an extra condition which says it is not an error if `QueueURL` is present. I have rolled back other test where `PathStyle` is `true` , `QueueURL` is `""` , `NonAWSBucketName ` is `""`

File: x-pack/filebeat/input/awss3/config.go (hunk 1 above)
Comment (bhapas) [reply x-pack/filebeat/input/awss3/config.go:106]: I think this has to be a little different now.

`path_style can only be used when using non-AWS S3 services or SQS/SNS QueueURL`

**Review by zmoog — APPROVED** [review]
LGTM.

It seems we're using endpoint resolution [v1](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/endpoints/#v1-endpointresolver), please correct me if I'm wrong.

This is fine if this is the case since AWS released [v2](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/endpoints/#v2-endpointresolverv2--baseendpoint) recently. Are there considerations or plans for v1 vs. v2?

**Comment by bhapas** [comment]
> LGTM.
> 
> It seems we're using endpoint resolution [v1](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/endpoints/#v1-endpointresolver), please correct me if I'm wrong.
> 
> This is fine if this is the case since AWS released [v2](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/endpoints/#v2-endpointresolverv2--baseendpoint) recently. Are there considerations or plans for v1 vs. v2?

[V2](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/endpoints/#v2-endpointresolverv2--baseendpoint) sounds more practical for testing purposes , specially `Localstack` . Probably would revisit this when the V2 is officially out in Q3 2023

**Pull request merged**